src/components/encoding-pane/function-picker.tsx

```diff
@@ -43,7 +61,12 @@ export class FunctionPickerBase extends React.PureComponent<FunctionPickerProps,
     }
   }
   private onFunctionChange(event: any) {
-    this.props.onFunctionChange(event.target.value as ShelfFunction);
+    const {index} = this.props;
+    if (index !== undefined) {
+      this.props.onFunctionChange(event.target.value, index);
```

Comment:
A better design of a reusable component would minimize unnecessary parameterization.  
Thus, I wonder if there is a way to structure this without requiring `index`.  

(At least, the any encoding shelf, which similarly is a list, also does not require index in `onFunctionChange`.)